New module: GitHub Codespaces

a-a-ron opened the pull request:
**Module title**: Start coding instantly with GitHub Codespaces

**Learning objectives**

- 3.5.1 Describe what is GitHub Codespaces
- 3.5.2 How do you start a GitHub codespace
- 3.5.3 Describe the [codespace lifecycle](https://docs.github.com/en/codespaces/getting-started/the-codespace-lifecycle) and how to do each step
- 3.5.4 Describe the different customizations you can personalize with GitHub Codespaces
- 3.5.5 Explain how to add and configure dev containers (keep high level, only understanding, not application)
- 3.5.6 Add an "Open in GitHub Codespaces" badge to a README file
- 3.5.7 Explain how to use the github.dev editor
- 3.5.8 Explain the differences between the github.dev editor and a GitHub Codespace

**Resources**
- https://docs.github.com/en/codespaces/overview

### Starting Module outline

![Screenshot 2023-05-24 at 9 16 39 AM](https://github.com/githubpartners/microsoft-learn/assets/6351798/ad7ebb10-91c3-4a56-9b5a-247b85812255)

### Action items

- [ ] Create a module outline from the template outline provided
  - This module will use the [codespaces skills exercise](https://github.com/skills/code-with-codespaces) for the exercise unit
- [ ] Peer review and final outline approved by @camihmerhar and @a-a-ron 
- [ ] Module first draft
- [ ] Module peer review
- [ ] Final module content approved by @camihmerhar and @a-a-ron

a-a-ron committed:
Add scaffolding for codespaces module

a-a-ron committed:
Remove unit

kfeeman1969 commented:
@a-a-ron Additional possible resource material includes:

- https://docs.github.com/en/codespaces/getting-started
- Blog Posts

I met with Cami today and discussed two things. One, how she has gone about working on a PR for a module and two, we went over an extremely rough outline. After our discussion, I think I should be able to have a draft outline for you and Cami to review EOD 6/5/2023

camihmerhar committed:
Update 1-introduction.md

camihmerhar committed:
Create 7-learning-content.md

camihmerhar committed:
Create 7-learning-content.yml

camihmerhar committed:
Update 7-learning-content.yml

kfeeman1969 committed:
Rename 2-learning-content.md to 2-Codespace-Lifecycle.md

Changed heading to match new content.

kfeeman1969 committed:
Rename 2-learning-content.yml to 2-Codespace-Lifecycle.yml

Changed file name to reflect content

kfeeman1969 committed:
Rename 2-Codespace-Lifecycle.md to 2-codespace-lifecycle.md

kfeeman1969 committed:
Rename 2-Codespace-Lifecycle.yml to 2-codespace-lifecycle.yml

kfeeman1969 committed:
Update 2-codespace-lifecycle.md

New content for content lifecycle.

kfeeman1969 committed:
Rename 4-learning-content.md to 3-managing.codespace.md

Change .md file to reflect content,

kfeeman1969 committed:
Rename 4-learning-content.yml to 3-managing.codespace..yml

Renamed .yml file to reflect the .md name change to keep consistent

kfeeman1969 committed:
Rename 3-managing.codespace..yml to 3-managing.codespace.yml

kfeeman1969 committed:
Update 3-managing.codespace.md

kfeeman1969 committed:
Delete 7-learning-content.md

Didn't need this file

kfeeman1969 committed:
Rename 5-exercise.md to 4-exercise.md

change file name to reflect outline

kfeeman1969 committed:
Rename 5-exercise.yml to 4-exercise.yml

kfeeman1969 committed:
Delete 7-learning-content.yml

No longer needed this file.

kfeeman1969 committed:
Create 5-knowledge-check.md

kfeeman1969 committed:
Rename 7-summary.md to 6-summary.md

kfeeman1969 committed:
Update 4-exercise.md

kfeeman1969 committed:
Update 6-summary.md

kfeeman1969 committed:
Rename 6-knowledge-check.yml to 5-knowledge-check.yml

kfeeman1969 committed:
Rename 7-summary.yml to 6-summary.yml

kfeeman1969 committed:
Update 4-exercise.md

kfeeman1969 committed:
Update 4-exercise.md

Updated link to existing exercise.

kfeeman1969 committed:
Update 4-exercise.md

Updated exercise file to contain CodeSpace information

kfeeman1969 committed:
Update 4-exercise.md

kfeeman1969 committed:
Update 4-exercise.md

kfeeman1969 committed:
Update 4-exercise.md

a-a-ron committed:
Small review changes to introduction.mg

a-a-ron committed:
Add images to 2-codespaces-lifecycle

a-a-ron committed:
Add changes

a-a-ron commented:
Hey @kfeeman1969! Great module! I made a couple changes while reviewing the content but first review was very nice!

After reading through the content, I feel like we mention github.dev too early and it feels a bit out of place. I think we need to move it out of the lifecycle unit and perhaps into it's own unit right before the exercise. Let's chat through this during our team sync tomorrow. I also added three images to the module. Two of them I added to the content and the last one is an image for port forwarding if we want to use it.

kfeeman1969 committed:
Update 5-knowledge-check.yml

Added question one with content

kfeeman1969 committed:
Update 5-knowledge-check.yml

Updated key and distractors for Question 1

kfeeman1969 committed:
Update 5-knowledge-check.yml

Entered knowledge check question 2 with stem, key, and distractors.

kfeeman1969 committed:
Update 5-knowledge-check.yml

kfeeman1969 committed:
Update 5-knowledge-check.yml

kfeeman1969 committed:
Update 2-codespace-lifecycle.md

Deleted github.dev information

kfeeman1969 committed:
Rename 3-managing.codespace.md to 4-managing.codespace.md

kfeeman1969 committed:
Rename 4-exercise.md to 5-exercise.md

kfeeman1969 committed:
Rename 5-exercise.md to 6-exercise.md

kfeeman1969 committed:
Rename 4-managing.codespace.md to 3-managing.codespace.md

kfeeman1969 committed:
Create 4-githubdev.md

Added new unit

kfeeman1969 committed:
Rename 6-summary.md to 7-summary.md

kfeeman1969 committed:
Rename 4-exercise.yml to 5-exercise.yml

kfeeman1969 committed:
Rename 5-knowledge-check.yml to 6-knowledge-check.yml

kfeeman1969 committed:
Rename 6-summary.yml to 7-summary.yml

kfeeman1969 committed:
Create 4-gethubdev.md

kfeeman1969 committed:
Update 6-knowledge-check.yml

kfeeman1969 committed:
Update 6-knowledge-check.yml

kfeeman1969 committed:
Update 6-knowledge-check.yml

kfeeman1969 committed:
Update 6-knowledge-check.yml

kfeeman1969 committed:
Update 6-knowledge-check.yml

kfeeman1969 committed:
Update 6-knowledge-check.yml

kfeeman1969 committed:
Update 6-knowledge-check.yml

kfeeman1969 committed:
Update and rename 4-gethubdev.md to 4-gethubdev.yml

camihmerhar committed:
Update 6-knowledge-check.yml

Changed some of the explanations of the the incorrect questions that way it didn't give away the answer 

Removed the 4th question, since we're trying to implement the standard to be the distractors have to be 100% possible but 100% incorrect we'll need to rework the distractors to work

camihmerhar committed:
Update 1-introduction.md

You typically don't need a header in the intro paragraph in the intro page because it'll provide a header for you.

Looked back at a previous module, it looks like we don't a colon in the Learning Objectives header.

kfeeman1969 committed:
Create 4-about-enterprise-managed-users.md

kfeeman1969 committed:
Rename 4-authorization.md to 5-authorization.md

kfeeman1969 committed:
Rename 5-team-synchronization.md to 7-team-synchronization.md

kfeeman1969 committed:
Rename 7-summary.yml to 8-summary.yml

kfeeman1969 committed:
Rename 7-team-synchronization.md to 8-team-synchronization.md

camihmerhar committed:
Update 1-introduction.md

Broke up the introduction paragraph to make it easier for readers to digest and edited learning objectives to align to learning units

camihmerhar committed:
Update and rename 4-githubdev.md to 4-githubdev-versus-GitHub-Codespaces.md

Removed introduction header for a more seamless experience with the updated name of the unit.

Also added specific nod to important content for the upcoming foundations exam

camihmerhar committed:
Update and rename 4-gethubdev.yml to 4-githubdev-versus-GitHub-Codespaces.yml

Updated minutes and title

camihmerhar committed:
Update 2-codespace-lifecycle.md

added content around resuming an existing codespace to align with the content on the foundation exam

camihmerhar committed:
Update 3-managing.codespace.md

Moving to 2nd unit in order to fit with flow

camihmerhar committed:
Update 2-codespace-lifecycle.md

Added in Losing internet connection while using Codespaces

camihmerhar committed:
Update 2-codespace-lifecycle.md

specified internet connection

camihmerhar committed:
Update 2-codespace-lifecycle.md

Removed a couple of sections to reduce the noise of the learning unit and pointed learners to more in depth material in the resources section

camihmerhar committed:
Update 2-codespace-lifecycle.md

updated headers to better delineate content

camihmerhar committed:
Update and rename 3-managing.codespace.md to 3-personalizing-your-codespace.md

Reframing this section to be shorter and aligned to foundations exam, placeholders for now

camihmerhar committed:
Update 3-personalizing-your-codespace.md

Updated content to include information on the foundations exam

camihmerhar committed:
Update 1-introduction.md

Removed 2nd prerequisite

camihmerhar committed:
Delete 5-knowledge-check.md

Deleting this file since we don't need it

camihmerhar committed:
Update 7-summary.md

Added content to summary

camihmerhar committed:
Update 3-managing.codespace.yml

Changed time stamp

camihmerhar committed:
Update 4-githubdev-versus-GitHub-Codespaces.md

Added a bit of content to ensure we cover questions reviewed in the foundations exam

a-a-ron committed:
Make minor changes

a-a-ron committed:
Merge branch 'main' into github-codespaces

a-a-ron committed:
Revert "Rename 7-team-synchronization.md to 8-team-synchronization.md"

This reverts commit e6b0fc374100e3321fc69169e10d2c0394aa7277.

a-a-ron committed:
Revert "Create 4-about-enterprise-managed-users.md"

This reverts commit 9798c9d12d9b951dcbc6b1016e2ea1bd54639788.

a-a-ron commented:
Hey @camihmerhar, this is almost ready to ship - just one last comment.

At the beginning of unit **4-githubdev-versus-GitHub-Codespaces.md** we jump right into github.dev with the following sentence. "Similar to Codespaces, you can also utilize github.dev". I feel like we need an introduction to github.dev. Nothing big, maybe a paragraph or two. Something to initiate the topic. 

Other than that, we are good to go! Thanks!!

camihmerhar committed:
Update 4-githubdev-versus-GitHub-Codespaces.md

added transition

camihmerhar committed:
Update 1-introduction.md

Updated last learning objective

camihmerhar committed:
Update 4-githubdev-versus-GitHub-Codespaces.md

Reworked the transition into the unit and provided clearer differentiation between the two products.

camihmerhar committed:
Update 4-githubdev-versus-GitHub-Codespaces.md

fixed "personal" typo

camihmerhar committed:
Update 4-githubdev-versus-GitHub-Codespaces.md

defined the transition sentence into the table to say codespaces and github.dev in order to be a bit clearer to the reader

camihmerhar committed:
Update 6-exercise.md

Implemented truncated version of exercise unit
